### PR TITLE
EVAL support in Dynomite

### PR DESCRIPTION
**Pull request by shailesh33**
Parse the EVAL request correctly. Also, restrict the keys in the script to belong to
same node.

### CODE REVIEW

File: src/dyn_message.h [hunk 1]
```diff
@@ -261,6 +265,8 @@ dn_strerror(dyn_error_t err)
             return "Datastore refused connection";
         case DYNOMITE_NO_QUORUM_ACHIEVED:
             return "Failed to achieve Quorum";
+        case DYNOMITE_SCRIPT_SPANS_NODES:
+            return "Script Spans multiple Dynomite nodes";
```
Comment (ipapapa) [inline src/dyn_message.h]: Usually, a failure message contains some negativity like "refused" or "failed". In this error message, I am not sure the client user will understand what it means.

Comment (shailesh33) [reply src/dyn_message.h]: good catch!. I will edit it as ```Keys in the script cannot span multiple Dynomite nodes```

File: src/proto/dyn_redis.c [hunk 2]
```diff
@@ -2938,6 +2951,31 @@ redis_fragment(struct msg *r, struct server_pool *pool, struct rack *rack, struc
     }
 }
 
+rstatus_t
+redis_verify_request(struct msg *r, struct server_pool *pool, struct rack *rack)
```
Comment (ipapapa) [inline src/proto/dyn_redis.c:2955]: `redis_verify_request` is very generic. Should it contain something about this being about LUA. At least a comment or some indication for the maintainer.

Comment (shailesh33) [reply src/proto/dyn_redis.c:2955]: There is a comment in the code. Also this is a general function to verify every request, some are handled, some are now. Just like redis_fragment,

File: src/dyn_message.h (hunk 1 above)
Comment (shailesh33) [reply src/dyn_message.h]: Fixed it. Now it looks like this:

```
127.0.0.1:8102> eval "return {redis.call('get',KEYS[1]),ARGV[1],ARGV[2]}" 3 key:000000058729 key:000000091650 key:000000019267 arg1 arg2
(error) ERR Dynomite: Keys in the script cannot span multiple nodes
```

**Pull request merged**